forgerock-authenticator/src/main/java/org/forgerock/android/auth/policy/FRAPolicy.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2023 ForgeRock. All rights reserved.
+ *
+ * This software may be modified and distributed under the terms
+ * of the MIT license. See the LICENSE file for details.
+ */
+
+package org.forgerock.android.auth.policy;
+
+import android.content.Context;
+
+import org.json.JSONObject;
+
+/**
+ * The FRAPolicy is an abstract Policy that provides general guidance on implementing policies
+ * to enforce the security of an Authenticator app.
+ *
+ * A policy must contain an unique name and instructions to determinate whether a condition is
+ * valid at a particular time. The policy may optionally contain some data to be used in the
+ * validation procedure.
+ */
+public abstract class FRAPolicy {
+
+    JSONObject data;
+
+    /**
+     * Set the attributes used for policy validation.
+     *
+     * @param data The attributes
+     */
+    public void setData(JSONObject data) {
+        this.data = data;
+    }
+
+    /**
+     * Retrieve the name of the Policy.
+     *
+     * @return The name of the Policy
+     */
+    public abstract String getName();
+
+    /**
+     * Evaluate the policy compliance.
+     *
+     * @param context  The Application Context.
+     * @return {@code true} if the policy comply, {@code false} otherwise
+     */
+    public abstract boolean evaluate(Context context);
```

Comment:
Something to think of not necessary to change it.
We may want to make the evaluate interface async like evaluate(context, FRListener), for example, if we want to have LocationAwarePolicy, getting the location from device is using callback, just in case we need to perform something and waiting for input.

Reply:
That's a good point. I will think about that.